### PR TITLE
Add SourceManager tests for Run and Wait methods

### PR DESCRIPTION
**Pull request by mcastorina**

### CODE REVIEW

File: pkg/sources/source_manager.go [hunk 1]
```diff
@@ -77,16 +87,14 @@ func (s *SourceManager) Enroll(ctx context.Context, name string, kind sourcespb.
 		return 0, fmt.Errorf("handle ID '%d' already in use", handleID)
 	}
 	s.handles[handleID] = f
-	return 0, nil
```
Comment (rosecodym) [inline pkg/sources/source_manager.go:80]: oops

**Review by rosecodym — APPROVED** [review]
lgtm but you probably should have a go-knower review the new `Wait` method

File: pkg/sources/source_manager.go (hunk 1 above)
Comment (mcastorina) [reply pkg/sources/source_manager.go:80]: Tests are great :)

File: pkg/sources/source_manager_test.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+func (c *counterChunker) Chunks(_ context.Context, ch chan *Chunk) error {
+	for i := 0; i < c.count; i++ {
+		ch <- &Chunk{Data: []byte{c.chunkCounter}}
+		c.chunkCounter++
+	}
+	return nil
+}
+
+// enrollDummy is a helper function to enroll a DummySource with a SourceManager.
+func enrollDummy(man *SourceManager, chunkMethod chunker) (handle, error) {
```
Comment (ahrav) [inline pkg/sources/source_manager_test.go:49]: Super nit: I feel like mgr reads a bit nicer than man for the manager.

**Review by ahrav — APPROVED** [review]
Very nice! Tests ftw

**Pull request merged**